torch/testing/_comparison.py

```diff
@@ -771,42 +773,59 @@ def _compare_sparse_coo_values(
             identifier="Sparse COO values",
         )
 
-    def _compare_sparse_csr_values(
+    def _compare_sparse_compressed_values(
         self, actual: torch.Tensor, expected: torch.Tensor, *, rtol: float, atol: float, equal_nan: bool
     ) -> None:
-        """Compares sparse CSR tensors by comparing
+        """Compares sparse compressed tensors by comparing
 
         - the number of non-zero elements (nnz) for equality,
-        - the col_indices for equality,
-        - the crow_indices for equality, and
+        - the plain indices for equality,
+        - the compressed indices for equality, and
         - the values for closeness.
         """
+        format_name = {
+            torch.sparse_csr: 'CSR',
+            torch.sparse_csc: 'CSC',
+            torch.sparse_bsr: 'BSR',
+            torch.sparse_bsc: 'BSC'}[actual.layout]
+
         if actual._nnz() != expected._nnz():
             raise self._make_error_meta(
                 AssertionError,
                 (
-                    f"The number of specified values in sparse CSR tensors does not match: "
+                    f"The number of specified values in sparse {format_name} tensors does not match: "
                     f"{actual._nnz()} != {expected._nnz()}"
                 ),
             )
 
+        compressed_indices_mth = {
+            torch.sparse_csr: torch.Tensor.crow_indices,
+            torch.sparse_csc: torch.Tensor.ccol_indices,
+            torch.sparse_bsr: torch.Tensor.crow_indices,
+            torch.sparse_bsc: torch.Tensor.ccol_indices}[actual.layout]
+        plain_indices_mth = {
+            torch.sparse_csr: torch.Tensor.col_indices,
+            torch.sparse_csc: torch.Tensor.row_indices,
+            torch.sparse_bsr: torch.Tensor.col_indices,
+            torch.sparse_bsc: torch.Tensor.row_indices}[actual.layout]
+
         self._compare_regular_values_equal(
-            actual.crow_indices(),
-            expected.crow_indices(),
-            identifier="Sparse CSR crow_indices",
+            compressed_indices_mth(actual),
```

Comment:
Is `mth` a sparse term? Otherwise, could you explain or replace with something more expressive?

Reply:
`mth` is an abbreviation for `method`.

Reply:
Ah :slightly_smiling_face: IMO the extra three characters will not hinder readability here and might help someone else reading this to not stumble over the same. No strong opinion though.

Reply:
Using `method`.